tools/istio-iptables/pkg/dependencies/implementation.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dependencies
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"os/user"
+	"strings"
+
+	"istio.io/pkg/env"
+)
+
+type RealDependencies struct {
+}
+
+func (r *RealDependencies) GetLocalIP() (net.IP, error) {
```

Comment:
all func's should have a comment header.

Reply:
Done